### ISSUE TITLE
Variable juice pricing - Congestion fees

### ISSUE THREAD

**Issue by mikera** [Convex-Dev/convex#423]
Juice price should be increased in case of high network load

Considerations:
- Should probably measure congestion in moving average juice per second
- Voting by peers undesirable - perverse incentives

**Comment by helins** [comment]
Juice per second relates to the CVM. However, signature validation is very expensive so the number of transactions and blocks should also be factored in to reflect that.

I.e. for the same amount of juice / sec, a higher volume of transactions will generate more load.

**Comment by mikera** [comment]
I believe a constant up-front juice per transaction is required to account at least partially for signature cost. This would also encourage bunding more complex / multiple transactions with a single signature.

**Comment by helins** [comment]
Probably best to factor-in blocks as well.
There must be an optimal block size, worst case being blocks with only 1 transaction.

**Comment by mikera** [comment]
Indeed, there should be a per-block charge to the submitting Peer I think to discourage too many small blocks
